p5/sketch/renderer3d.py

```diff
@@ -141,11 +159,12 @@ def render(self, shape):
 				np.hstack([shape.vertices, np.ones((n, 1))]),
 				shape.matrix,
 				self.transform_matrix)
+			tnormals = shape.vertex_normals @ np.linalg.inv(to_3x3(self.transform_matrix) @ to_3x3(shape.matrix))
```

Comment:
Would having a different `render` function for different materials be more efficient?

Reply:
That's a good point. I'll think about how to implement it.